services/api/src/utils/middleware/tokens.js

```diff
@@ -62,6 +74,10 @@ function getToken(ctx) {
   return token;
 }
 
+function getParamsToken(ctx) {
+  return ctx.request.query?.token || ctx.request.body?.token;
```

Comment:
ctx.request.query?.token should not be preferred way to deal with tokens, specially if they are long lasting, as query stored in logs + browser history etc.

Reply:
They are not long lasting this is for one-off tokens like the mail token which are short lived.